packages/mock-block-dock/dev/test-react-block.tsx

```diff
@@ -2,39 +2,63 @@ import {
   type BlockComponent,
   useGraphBlockService,
 } from "@blockprotocol/graph/react";
+import { getRoots } from "@blockprotocol/graph/stdlib";
 import { useHook, useHookBlockService } from "@blockprotocol/hook/react";
-import { useRef } from "react";
+import { extractBaseUri } from "@blockprotocol/type-system/slim";
+import { useMemo, useRef } from "react";
+
+import { propertyTypes } from "../src/data/property-types";
 
 type AppProps = {
   name: string;
 };
 
-export const TestReactBlock: BlockComponent<AppProps> = ({ graph }) => {
-  const {
-    blockEntity: { entityId, properties },
-    readonly,
-  } = graph;
+export const TestReactBlock: BlockComponent = ({ graph }) => {
+  const { blockEntitySubgraph, readonly } = graph;
   const blockRef = useRef<HTMLDivElement>(null);
   const hookRef = useRef<HTMLDivElement>(null);
 
+  const blockEntity = useMemo(() => {
+    return blockEntitySubgraph ? getRoots(blockEntitySubgraph)[0]! : undefined;
+  }, [blockEntitySubgraph]);
+
   const { graphService } = useGraphBlockService(blockRef);
 
   const { hookService } = useHookBlockService(blockRef);
 
-  useHook(hookService, hookRef, "text", entityId, "$.description", () => {
-    throw new Error(
-      "Fallback called – dock is not correctly handling text hook.",
-    );
-  });
+  useHook(
+    hookService,
+    hookRef,
+    "text",
+    blockEntity?.metadata.editionId.baseId ?? "",
+    "$.description",
+    () => {
+      throw new Error(
+        "Fallback called – dock is not correctly handling text hook.",
+      );
+    },
+  );
 
   if (readonly) {
     return (
       <div ref={blockRef}>
         <h1>
-          Hello {properties.name}! The id of this block is {entityId}
+          Hello{" "}
+          {
+            blockEntity?.properties[extractBaseUri(propertyTypes.name.$id)] as
+              | string
+              | undefined
```

Comment:
This is the sort of thing that should become a lot more ergonomic with type generation